adagrams/game.py

```diff
@@ -1,11 +1,125 @@
+import random
+
 def draw_letters():
-    pass
+    letters = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    hand = []
+    for i in range(10):
+        choose_letter = list(letters.keys())
+        letter_choice = random.choice(choose_letter)
```

Comment:
Nice job using a built-in function! How does random.choice() work under the hood?

I encourage students to practice writing this kind of logic on their own because it provides practice in thinking through the algorithm to grab a random letter and provides the muscle memory of writing this kind of code too. 

While writing it out on your own is more lines of code, I think that's ok for initial projects to you get this practice.

Especially in interviews, we should feel confident that we understand how built-in functions work under the hood because your interviewer may ask you to explain it to them.

Reply:
Ofc it's totally fine to use built in functions, why reinvent the wheel. It's just that we want to encourage folks to think through how things work under the hood so when it comes to interviews you're ready to explain what you use. So the best practice for that is implementing it yourself. One way to randomly grab a letter from a list:

```python 
 random_int = random.randint(0, len(letter_pool) - 1)
 random_letter = pool[random_int]
```

where letter_pool is a list with all the available tiles to draw from ["A", "A", "A", "A", "A", "A", "A", "A", "A", "B", "B"...]

Reply:
Here I was creating said helper function. 🤦🏽‍♀️ Don't worry about responding; I know it's getting late and I really appreciate all your help. I can plan to go to office hours next week. I should probably be working on the PSE or getting a handle on git or debugging anyway. I guess I just still don't understand the difference, and I don't think I entirely understand what "under the hood" means.
![image](https://user-images.githubusercontent.com/123524891/227664248-ec48f447-fdb3-4cc9-b386-0cee1022d611.png)